### ISSUE TITLE
[PROJECT] Improve the accessibility of the platform

### ISSUE THREAD

**Issue by Memo-Es** [opencollective/opencollective#4045]
## Background

With the platform's growth so far and considering our plans for the short and long-term future, prioritizing accessibility is crucial for its correct evolution. By enabling more users to use our product, we align our mission of unlocking economic power to more projects and initiatives. It substantially improves the experience of using the platform for normal users across all devices and screen sizes.

## Project goal

Scan the platform's design and make the necessary improvements to comply with the [WCAG standard](https://www.w3.org/TR/WCAG20/) for text readability and usability different platform features.

## Team

@Memo-Es (Owner)
@Betree (Engineering)

## Project priorities

- Comply with the [WCAG standards](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) for text readability platform-wide
- Do the necessary modifications to the design system, resulting in a series of updates to be published as the main deliverable of this project
- Audit the UI components like buttons, form elements, cards, etc. to ensure they help deliver the best experience possible

**Note:** We may leave some of the findings we encounter in the process to be tacked in a future accessibility project or on a sidetrack.

## References and links

- [Kick-off meeting notes](https://docs.google.com/document/d/10tT1MW6lIIuJBxbNHMHTnEf9HDsFxYZUTKTke5ZuM6E/edit#)
- **GitHub issues**
- #3943 #3528 
- Freshdesk ticket about mobile accessibility: https://opencollective.freshdesk.com/a/tickets/14104

## Progress tracking

This project will focus mainly on weighing the importance of designing the platform with clear standards that help us maintain a consistent product evolution, eventually setting up an industry standard for visual accessibility online.
- We will start with the design system's atomic level, along with the main components most widely used by our users.

### Milestones
- [x] [Buttons compliance](https://github.com/opencollective/opencollective/issues/4045#issuecomment-814581377)
- [x] [Form fields compliance](https://github.com/opencollective/opencollective/issues/4045#issuecomment-814603020)
- [x] [Tier cards compliance](https://github.com/opencollective/opencollective/issues/4045#issuecomment-814614922)
- [x] [Transactions items](https://github.com/opencollective/opencollective/issues/4045#issuecomment-815151795)
- [x] [Global navigation](https://github.com/opencollective/opencollective/issues/4045#issuecomment-817450685)
- [x] [Collective page](https://github.com/opencollective/opencollective/issues/4045#issuecomment-818403648)
- [ ] Accessibility documentation

**Comment by Betree** [comment]
On the engineering side, we also have to work on the metadata (`aria-label`, etc) and alternatives text for images (in particular to improve the experience with screen readers). @SudharakaP introduced a tool in the codebase that will help to enforce good practices in the future.

**Comment by SudharakaP** [comment]
> On the engineering side, we also have to work on the metadata (`aria-label`, etc) and alternatives text for images (in particular to improve the experience with screen readers). @SudharakaP introduced a tool in the codebase that will help to enforce good practices in the future.

Yes, I wasn't sure if this falls part of this project or not, but I am working on the remainder of the accessibility issues reported by that tool here; https://github.com/opencollective/opencollective-frontend/pull/596. Will be done by end of this week. 😄

**Comment by SudharakaP** [comment]
As a side note, I was investigating some accessibility related issues on the site specifically working on the aforementioned PR, and I found that the following extension (which uses [axe-core](https://github.com/dequelabs/axe-core)) might be quite helpful in identifying accessibility issues in our DOM; 

https://www.deque.com/axe/browser-extensions/. 

For each page it will list accessibility problems along with the potential fixes we could do. I think this might be specially useful to identify the color contrasting issues; for example in the homepage there's around 10 of those. 😄 

![image](https://user-images.githubusercontent.com/12435965/110193650-0aaf1e00-7dea-11eb-938a-b2bd12ce6ef3.png)

**Comment by Memo-Es** [comment]
## Buttons

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- Added the 'Action Button' for profile pages (Collective page navigation redesign project)

### Preview

![❖  Platform Buttons](https://user-images.githubusercontent.com/49539954/113808086-e0bf8300-972a-11eb-970f-8fc0ddeae789.png)

## [🎨 → Link to figma file](https://www.figma.com/file/N4Xbl652BhzOutXmzhcrLGch/DS-02-Atoms?node-id=3842%3A222)

**Comment by Memo-Es** [comment]
## Form Fields

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- States variants for text field, text area, and rich text field added
- Update to the hint text for the text component, adding a visual indicator of the field state

### Preview

![❖ Form Fields](https://user-images.githubusercontent.com/49539954/113813021-b4a8ff80-9734-11eb-92af-371fd87407a3.png)


## [🎨 → Link to figma file](https://www.figma.com/file/N4Xbl652BhzOutXmzhcrLGch/DS-02-Atoms?node-id=3050%3A68)

**Comment by Memo-Es** [comment]
## Tier Cards

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- States variants for the card interaction – Card hover, tier card title, read more (Both the title and the read more text links take you to the standalone tier page)

### Preview

![❖ Tier Cards](https://user-images.githubusercontent.com/49539954/113815473-d1dfcd00-9738-11eb-86f3-93d31e5b6650.png)


## [🎨 → Link to figma file](https://www.figma.com/file/9Iq3vBOmZ2EuOacXEYD5yK/DS-03-Global-Components?node-id=6253%3A84)

**Comment by Memo-Es** [comment]
## Transactions Components

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- States variants for the transaction list interaction – Element on hover, expense title, read more (Expense title leads to the standalone expense page)

### Preview

![❖ Design Masters](https://user-images.githubusercontent.com/49539954/113919871-b1039000-97a9-11eb-998e-02339c410711.png)

## [🎨 → Link to figma file](https://www.figma.com/file/9Iq3vBOmZ2EuOacXEYD5yK/DS-03-Global-Components?node-id=5165%3A0)

**Comment by kaylarep** [comment]
Sorry am late to adding to this issue, but there are also some suggestions from [this Freshdesk ticket](https://opencollective.freshdesk.com/a/tickets/10885)

> your website can be made more accessible to folks with minor vision impairment, and to the very large percent of the population that was born before folks used computers and cell phones for all their business - which wasn't that long ago :)

> The most basic observation is that fonts should be larger, and text should be black rather than gray.  For anyone without perfect vision, it can be very hard to read gray text, no matter now large.  While it's possible for a user to enlarge fonts on any website, that often leads to crucial info dropping off the screen.  Starting with a more accessible format would make things much easier.

**Comment by Memo-Es** [comment]
## Global Navigation

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- Update on the hover state for text links (Underlined)
- Update the nav bar and footer
- Update the design of the menu on both desktop+tablet and mobile

### Preview
![❖  Platform Navigation](https://user-images.githubusercontent.com/49539954/114336022-ae8b9800-9b13-11eb-82ea-ab409dac3479.png)


## [🎨 → Link to figma file](https://www.figma.com/file/9Iq3vBOmZ2EuOacXEYD5yK/DS-03-Global-Components?node-id=4%3A0)

**Comment by Memo-Es** [comment]
## Collective Navigation

### Description of changes
- Color correction to comply with the 4:5:1 ratio
- Text styles update
- Removed the 'Settings' button from the collective navbar and placed it in its original position below the avatar (For reference on the action button full roll-out check the [buttons update](https://github.com/opencollective/opencollective/issues/4045#issuecomment-814581377).
- Update the Collective navbar component to leave only the action buttons aimed to interact with that Collective
- Update to the desktop menu for the 'Actions' dropdown button
- Update the mobile navigation experience including the new design for the action buttons on mobile
- Implement the new horizontal navigation for tier cards described in https://github.com/opencollective/opencollective/issues/3783
- **Note:** Please go through all the page, there are small text style corrections and modifications that need to be included in this update 

### Preview
![Desktop](https://user-images.githubusercontent.com/49539954/114492265-dba47d00-9bdd-11eb-8441-e94b2390f512.png)


## [🎨 → Link to figma file](https://www.figma.com/file/e71tBo0Sr8J7R5n6iMkqI42d/OC.com-04-Collectives?node-id=0%3A7428)

**Comment by piamancini** [comment]
Review Call: 
 Not much progress in implementation. 
We need to divide each comment into separate issues and add a bounty on each one. 
@Betree to take up for Q2S2

**Comment by BenJam** [comment]
Prioritisation notes: not a major piece of work to tackle, quite well scoped and mostly limited to variable changes.

**Comment by Betree** [comment]
https://github.com/opencollective/opencollective-frontend/pull/6388 is going to resolve this issue. There are however some items that I've left unaddressed:

- Mobile user menu (will create an issue for this one)
- Contributors section - use larger avatars at the top: It's tricky to implement without breaking the responsiveness. I also think that we should discuss this change with the team.
- Collective page - section title colors: I couldn't find a clear pattern to describe the color of the section titles (eg. Updates is black.700 and Conversations is black.800)
- Goals: I'll create a separate issue, as we don't want to invest too much in maintaining this component at the moment
